aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -100,10 +100,11 @@ Tensor pinverse(const Tensor& self, double rcond) {
     return at::empty(self_sizes, self.options());
   }
   Tensor U, S, V;
-  std::tie(U, S, V) = self.svd();
+  std::tie(U, S, V) = self.conj().svd();
   Tensor max_val = at::narrow(S, /*dim=*/-1, /*start=*/0, /*length=*/1);
-  Tensor S_pseudoinv = at::where(S > rcond * max_val, S.reciprocal(), at::zeros({}, self.options()));
-  return at::matmul(V, at::matmul(S_pseudoinv.diag_embed(/*offset=*/0, /*dim1=*/-2, /*dim2=*/-1), U.transpose(-2, -1)));
+  Tensor S_pseudoinv = at::where(S > rcond * max_val, S.reciprocal(), at::zeros({}, S.options()));
+  Tensor S_pseudoinv_diag = S_pseudoinv.diag_embed(/*offset=*/0, /*dim1=*/-2, /*dim2=*/-1).to(self.dtype());
+  return at::matmul(V, at::matmul(S_pseudoinv_diag, U.transpose(-2, -1)));
```

Comment:
Hmm, could you please explain why just `U.transpose` and not `U.transpose.conj`?
The SVD decomposition of `self` is `self = U S V*`, so `pinv = V pseudo_reciprocal(S) U*`, as `V` and `U` are Hermitian.

Reply:
Ok, I see that you run `svd` on the conjugated result. But it could be confusing with this order and requires inserting some comments, maybe running svd directly on `self` is easier to understand?

Reply:
When I look at it now I agree with you that calling the svd directly on `self` is more clear.
I will change it to
```cpp
std::tie(U, S, V) = self.svd();
...
return at::matmul(V.conj(), at::matmul(S_pseudoinv_diag, U.transpose(-2, -1).conj()));
```

Reply:
Also, we can save on one matrix product and diag_embed by computing `U S V*` as ` (U * S) @ V*`.